chartfx-dataset/src/test/java/de/gsi/dataset/spi/DataSetBuilderTests.java

```diff
@@ -335,23 +360,146 @@ public void testMultiDimDataSet5() {
+            new DataSetBuilder().setNegError(DIM_X, new float[] { 0.0f, 3f }).setInitalCapacity(2).setDimension(2).build();
         });
+        assertThrows(UnsupportedOperationException.class, () -> {
+            new DataSetBuilder().setNegError(DIM_X, new double[] { 0.0, 3 }).setInitalCapacity(2).setDimension(2).build();
+        });
+        // Provide negative dimIndex for axisDescriptions
         assertThrows(UnsupportedOperationException.class, () -> {
             new DataSetBuilder().setAxisMax(-5, 12).build();
         });
+        // request float error data set
         assertThrows(UnsupportedOperationException.class, () -> {
             new DataSetBuilder().setUseFloat(true).setEnableErrors(true).build();
         });
+        // request errors for multi dim data set
         assertThrows(UnsupportedOperationException.class, () -> {
             new DataSetBuilder().setDimension(3).setEnableErrors(true).build();
         });
+        // request float multi dim data set
         assertThrows(UnsupportedOperationException.class, () -> {
             new DataSetBuilder().setDimension(3).setUseFloat(true).build();
         });
+        // request type without errors but add errors
+        assertThrows(UnsupportedOperationException.class, () -> {
+            new DataSetBuilder().setNegError(DIM_Y, new float[] { 1, 2 }).build(DefaultDataSet.class);
+        });
+        // request Editable Data Set
+        assertThrows(UnsupportedOperationException.class, () -> {
+            new DataSetBuilder() //
+                    .setValues(DIM_Y, new float[] { 1, 2 }) //
+                    .build(DoubleErrorDataSet.class);
+        });
+    }
+
+    @Test
+    public void testHelperFunctions() {
+        assertDoesNotThrow(() -> {
+            final DataSetBuilder builder = new DataSetBuilder("foo");
+            DataSet test0 = new MinimalDataSet();
+            builder.addMetaData(test0);
+            builder.setMetaErrorList("foo").setMetaInfoList("asdf", "qwert").setMetaWarningList("123", "321").setMetaInfoMap(Map.of("key", "value"));
+            DoubleErrorDataSet test1 = new DoubleErrorDataSet("test");
+            builder.addMetaData(test1);
+            builder.addMetaData(test0);
+        });
+        assertDoesNotThrow(() -> {
+            final DataSetBuilder builder = new DataSetBuilder("foo");
+            DataSet test0 = new MinimalDataSet();
+            builder.addDataLabelStyleMap(test0);
+            builder.setDataLabelMap(Map.of(1, "test", 2, "label")).setDataStyleMap(Map.of(1, "color:red"));
+            DoubleErrorDataSet test1 = new DoubleErrorDataSet("test", 2);
+            builder.addDataLabelStyleMap(test1);
+            builder.addDataLabelStyleMap(test0);
+        });
+    }
+
+    private static class MinimalDataSet implements DataSet {
+        private static final long serialVersionUID = 1L;
+        AtomicBoolean autoNotify = new AtomicBoolean();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5594966)